packages/Operators/test/tstNodeToNodeOperator.cpp

```diff
@@ -173,12 +175,30 @@ void setupAndRunTest( const std::string &input_file,
 //---------------------------------------------------------------------------//
 // Tests.
 //---------------------------------------------------------------------------//
-TEUCHOS_UNIT_TEST( NodeToNodeOperator, node_to_node_test )
+TEUCHOS_UNIT_TEST( NodeToNodeOperator, matching_node_test )
+{
+    // Run the test.
+    Teuchos::Array<double> gold_data;
+    Teuchos::Array<double> test_result;
+    setupAndRunTest( "matching_node_test.xml", gold_data, test_result, false );
+
+    // Check the results.
+    TEST_EQUALITY( gold_data.size(), test_result.size() );
+    int num_points = gold_data.size();
+    for ( int i = 0; i < num_points; ++i )
+    {
+        TEST_FLOATING_EQUALITY( gold_data[i], test_result[i], epsilon );
+    }
+}
+
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( NodeToNodeOperator, non_matching_node_test )
 {
     // Run the test.
     Teuchos::Array<double> gold_data;
     Teuchos::Array<double> test_result;
-    setupAndRunTest( "node_to_node_test.xml", gold_data, test_result );
+    setupAndRunTest( "non_matching_node_test.xml", gold_data, test_result,
+                     true );
```

Comment:
Does `Teuchos` have something like `BOOST_CHECK_THROW` that would catch the `DataTransferKitException`?  We do not trigger that in the test.